src/Maker/MakeWebhook.php

```diff
@@ -0,0 +1,285 @@
+        $dependencies->addClassDependency(
+            AbstractRequestParser::class,
+            'webhook'
+        );
+        $dependencies->addClassDependency(
+            ConsumerInterface::class,
+            'remote-event'
+        );
+        $dependencies->addClassDependency(
+            Yaml::class,
+            'yaml'
+        );
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        if ($input->getArgument('name')) {
+            if (!$this->verifyWebhookName($input->getArgument('name'))) {
+                throw new \InvalidArgumentException('A webhook name can only have alphanumeric characters, underscores, dots, and dashes.');
+            }
+
+            return;
+        }
+
+        $argument = $command->getDefinition()->getArgument('name');
+        $question = new Question($argument->getDescription());
+        $question->setValidator(Validator::notBlank(...));
+        $webhookName = $io->askQuestion($question);
+
+        while (!$this->verifyWebhookName($webhookName)) {
+            $io->error('A webhook name can only have alphanumeric characters, underscores, dots, and dashes.');
+            $webhookName = $io->askQuestion($question);
+        }
+
+        $input->setArgument('name', $webhookName);
+    }
+
+    private function verifyWebhookName(string $entityName): bool
+    {
+        return preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_.\-\x7f-\xff]*$/', $entityName);
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator): void
+    {
+        $webhookName = $input->getArgument('name');
+        $requestParserDetails = $this->generator->createClassNameDetails(
+            Str::asClassName($webhookName.'RequestParser'),
+            'Webhook\\'
+        );
+        $remoteEventHandlerDetails = $this->generator->createClassNameDetails(
+            Str::asClassName($webhookName.'WebhookHandler'),
+            'RemoteEvent\\'
+        );
+
+        $this->addToYamlConfig($webhookName, $requestParserDetails);
+
+        $this->generateRequestParser($io, $requestParserDetails);
+
+        $this->generator->generateClass(
+            $remoteEventHandlerDetails->getFullName(),
+            'webhook/WebhookHandler.tpl.php',
+            [
+                'webhook_name' => $webhookName,
+            ]
+        );
+
+        $this->generator->writeChanges();
+        $this->fileManager->dumpFile(self::WEBHOOK_CONFIG_PATH, $this->ysm->getContents());
+    }
+
+    private function addToYamlConfig(string $webhookName, ClassNameDetails $requestParserDetails): void
+    {
+        if (!$this->fileManager->fileExists(self::WEBHOOK_CONFIG_PATH)) {
+            $yamlConfig = Yaml::dump(['framework' => ['webhook' => ['routing' => []]]], 4, 2);
+        } else {
+            $yamlConfig = $this->fileManager->getFileContents(self::WEBHOOK_CONFIG_PATH);
+        }
+        $this->ysm = new YamlSourceManipulator($yamlConfig);
```

Comment:
Are we calling `$this->ysm` outside of this method? If not, let's remove the class property and keep this a local variable